nncf/openvino/graph/layer_attributes.py

```diff
@@ -67,9 +67,97 @@ def get_const_port_ids(self) -> List[int]:
             return list(self._constant_attributes.keys())
         return []
 
+    def get_backend_agnostic_attributes(self):
+        return self._layer_attributes
+
+
+OV_CONV_METATYPES = [
+    OVConvolutionMetatype,
+    OVDepthwiseConvolutionMetatype,
+    OVGroupConvolutionMetatype,
+    OVConvolutionBackpropDataMetatype,
+    OVGroupConvolutionBackpropDataMetatype,
+]
+
+
+def get_conv_weights_layout_from_node(node: NNCFNode) -> List[OVLayoutElem]:
+    """
+    Calculates weights layout for a target convolution node.
+
+    :param node: Target convolution node.
+    :return: Target convolution Node weights layout.
+    """
+    layer_attributes = node.layer_attributes
+    port_id = _get_constant_port_id_from_layer_attributes(layer_attributes)
+    return get_conv_weights_layout(
+        ov_metatype=node.metatype, weights_shape=layer_attributes.constant_attributes[port_id]["shape"]
+    )
+
+
+def get_linear_weights_layout_from_node(node: NNCFNode) -> List[OVLayoutElem]:
+    """
+    Calculates weights layout for a target linear node.
+
+    :param node: Target linear node.
+    :return: Target linear Node weight layout.
+    """
+    layer_attributes = node.layer_attributes
+    port_id = _get_constant_port_id_from_layer_attributes(layer_attributes)
+    constant_layer_attrs = layer_attributes.constant_attributes[port_id]
+    return get_linear_weights_layout(
+        weights_shape=constant_layer_attrs["shape"],
+        transpose=constant_layer_attrs["transpose"],
+        port_id=port_id,
+    )
+
+
+def _get_constant_port_id_from_layer_attributes(layer_attributes: OVLayerAttributes) -> int:
+    """
+    Returns constant ports id for convolutional and linear ops layer attributes.
+
+    :param layer_attributes: Target convolutional/linear layer op layer attributes.
+    :return: Constant port id for the target convolutional/linear model.
+    """
+    port_ids = list(layer_attributes.constant_attributes.keys())
+    assert len(port_ids) == 1
+    return port_ids[0]
+
+
+def get_conv_weights_layout(ov_metatype: OVOpMetatype, weights_shape: Tuple[int, ...]) -> List[OVLayoutElem]:
+    """
+    Calculates weights layout for a target convolution node.
+
+    :param ov_metatype: Target convolution node OpenVINO metatype.
+    :param weights_shape: Shape of the target convolution node weight.
+    :return: Target convolution node weights layout.
+    """
+    weights_layout = ov_metatype.const_layout
+    kernel_size = weights_shape[len(weights_layout) :]
+    weights_layout += [OVLayoutElem.SPATIAL] * len(kernel_size)
+    return tuple(weights_layout)
```

Comment:
Can we be sure that weight layout calculation is possible? For example, it is possible that between weights and the target layer, some reshapes are placed.